content/guides/use-case/wiremock.md

```diff
@@ -0,0 +1,313 @@
+
+   - The main file for the application is `index.js`, located in the `accuweather-api/src/api` directory.
+   - This file starts the `getWeather.js` module, which is essential for your Node.js application. It uses the `dotenv` package to load environment variables from the`.env` file.
+   - Based on the value of `API_ENDPOINT_BASE`, the application routes requests either to the WireMock server (`http://localhost:8080`) or the AccuWeather API. In this setup, it uses the WireMock server.
+   - The code ensures that the `ACCUWEATHER_API_KEY` is required only if the application is not using WireMock, enhancing efficiency and avoiding errors.
+
+
+    ```javascript
+    require("dotenv").config();
+
+    const express = require("express");
+    const axios = require("axios");
+
+    const router = express.Router();
+    const API_ENDPOINT_BASE = process.env.API_ENDPOINT_BASE;
+    const API_KEY = process.env.ACCUWEATHER_API_KEY;
+
+    console.log('API_ENDPOINT_BASE:', API_ENDPOINT_BASE);  // Log after it's defined
+    console.log('ACCUWEATHER_API_KEY:', API_KEY);          // Log after it's defined
+
+    if (!API_ENDPOINT_BASE) {
+      throw new Error("API_ENDPOINT_BASE is not defined in environment variables");
+    }
+
+    // Only check for API key if not using WireMock
+    if (API_ENDPOINT_BASE !== 'http://localhost:8080' && !API_KEY) {
+      throw new Error("ACCUWEATHER_API_KEY is not defined in environment variables");
+    }
+    // Function to fetch the location key for the city
+    async function fetchLocationKey(townName) {
+      const { data: locationData } = await
+    axios.get(`${API_ENDPOINT_BASE}/locations/v1/cities/search`, {
+        params: { q: townName, details: false, apikey: API_KEY },
+      });
+      return locationData[0]?.Key;
+    }
+    ```  
+
+    > [!TIP]
+    > For the production environment, it's recommended not to expose API keys details in logs. Instead, you can log a placeholder message to indicate whether the API key is loaded:
+
+      ```javascript
+      const API_KEY = process.env.ACCUWEATHER_API_KEY;
+
+      console.log('API_ENDPOINT_BASE:', API_ENDPOINT_BASE);  // Log after it's defined
+      console.log('ACCUWEATHER_API_KEY is set:', !!API_KEY); // Log a boolean value instead of the actual key
+
+      if (!API_ENDPOINT_BASE) {
+        throw new Error("API_ENDPOINT_BASE is not defined in environment variables");
+      }
+
+      // Only check for API key if not using WireMock
+      if (API_ENDPOINT_BASE !== 'http://localhost:8080' && !API_KEY) {
+        throw new Error("ACCUWEATHER_API_KEY is not defined in environment variables");
+      }
+      ```
+
+    > This approach ensures that you only log whether the key is set without revealing its value.
+
+4. Start the Node server
+
+   Before you start the Node server, ensure that you have already installed the node packages listed in the package.json file by running `npm install`. 
+
+   ```console
+   npm install 
+   npm run start
```

Comment:
This will fail on macOS because apple is already running a service on port 5000. Let's change the default port to `5001` in `accuweather-api/src/index.js`

Reply:
Makese sense. I went ahead and changed the port to 5001.